lib/routes/cool18/index.ts

```diff
@@ -50,17 +50,12 @@ async function handler(ctx: Context) {
     let list: any[];
 
     if (type === 'home') {
-        const scripts = $('script').toArray();
-        let pageData = [];
-        for (const script of scripts) {
-            const scriptContent = $(script).html();
-            const match = scriptContent?.match(/const\s+_PageData\s*=\s*(\[[\s\S]*?]);/);
-
-            if (match) {
-                pageData = JSON.parse(match[1]);
-            }
-        }
-        pageData.length = Number(pageSize);
+        const pageData = JSON.parse(
+            $('script:contains("_PageData")')
+                .text()
+                .match(/const\s+_PageData\s*=\s*(\[[\s\S]*?]);/)?.[1] || '[]'
+        );
+        pageData.length = ctx.req.query('limit') ? Number(ctx.req.query('limit')) : 20;
         list = pageData.map((item: PageDataItem) => ({
```

Comment:
[slice](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) the array instead of manipulating the length property.